cmd/meroxa/root/apps/deploy.go

```diff
@@ -200,17 +261,21 @@ func (d *Deploy) Execute(ctx context.Context) error {
 
 	switch d.lang {
 	case GoLang:
-		err = d.goDeploy.DeployGoApp(ctx, d.path, d.logger)
+		// The only reason Deploy is scoped this other way is, so we can have the Docker Credentials
+		// Maybe that function should take care of checking type of deployment, only passing flags
+		// and environment variables
+		// err = turbineGo.Deploy(ctx, d.path, d.logger)
+		deployOuput, err = d.goDeploy.Deploy(ctx, d.path, d.logger)
 	case "js", JavaScript, NodeJs:
-		err = turbineCLI.DeployJSApp(ctx, d.path, d.logger)
+		err = turbineJS.Deploy(ctx, d.path, d.logger)
 	default:
 		return fmt.Errorf("language %q not supported. %s", d.lang, LanguageNotSupportedError)
 	}
-
 	if err != nil {
 		return err
```

Comment:
If the deploy should fail with an error, would we need to log the error explicitly here via `logger.Error` or is this already taken care of by returning the `err` object?

Reply:
Got it, I wasn't fully aware yet, that an error essentially needs to be passed through to different commands and that sounds good!